contracts/bridge/x/v4/RegistrationV4.sol

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Immutable Pty Ltd 2018 - 2024
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import {CoreV4} from "./CoreV4.sol";
+
+contract RegistrationV4 {
+    CoreV4 public immutable imx;
+
+    constructor(address payable _imx) {
```

Comment:
I am pretty sure _imx does not need to be payable, and should hence, not be payable.

Reply:
got copied from v3, maybe because we were supporting deposits before. I'll remove it.

Reply:
The abi they provided us to generate this file has a payable fallback, is there any benefit here if we manually modify?

Reply:
let's not modify it - better to leave generated code as is.